pacman/executor/pacman_algorithm_executor.py

```diff
@@ -251,25 +290,28 @@ def _determine_algorithm_order(
         optionals_to_use = list(optional_algorithm_data)
         outputs_to_find = self._remove_outputs_which_are_inputs(
             required_outputs, inputs)
+        tokens_to_find = self._remove_complete_tokens(
```

Comment:
this is obsoleted, could be pushed into the building logic isntead of a seperate function thats only called once

Reply:
By that logic so could _remove_outputs_which_are_inputs could also be removed, but I think you will find that you added that one!